### PR TITLE
Onboard Argus Alert Config child routes in Route

### PR DESCRIPTION
**Pull request by DiogoFerrao**

### CODE REVIEW

File: stackit/internal/services/argus/instance/resource.go [hunk 1]
```diff
@@ -185,6 +199,111 @@ var webHooksConfigsTypes = map[string]attr.Type{
...
+// getRouteListType is a helper function to return the route list attribute type.
+func getRouteListType() types.ObjectType {
+	return getRouteListTypeAux(1, childRouteMaxRecursionLevel)
+}
+
+// getRouteListTypeAux returns the type of the route list attribute with the given level of child routes recursion.
+// The level is used to determine the current depth of the nested object.
+// The limit is used to determine the maximum depth of the nested object.
+// The level should be lower or equal to the limit, if higher, the function will produce a stack overflow.
+func getRouteListTypeAux(level, limit int) types.ObjectType {
```
[the remark this answers is not in the document]
Comment (DiogoFerrao) [reply stackit/internal/services/argus/instance/resource.go:222]: I was doing that previously, but this allows testing of the auxiliary function with different values, which makes sure that the method works for actual recursion (limit > 1)

File: stackit/internal/services/argus/instance/resource_test.go
[the remark this answers is not in the document]
Comment (DiogoFerrao) [reply stackit/internal/services/argus/instance/resource_test.go:1]: I'm working on it, currently having some problems with the latter

[the remark this answers is not in the document]
Comment (DiogoFerrao) [reply stackit/internal/services/argus/instance/resource_test.go:1]: Done now

File: stackit/internal/services/argus/argus_acc_test.go [hunk 2]
```diff
@@ -338,7 +363,7 @@ func TestAccResource(t *testing.T) {
 					utils.Ptr(instanceResource["metrics_retention_days"]),
 					utils.Ptr(instanceResource["metrics_retention_days_1h_downsampling"]),
 					utils.Ptr(instanceResource["metrics_retention_days_5m_downsampling"]),
-					buildAlertConfig(buildAlertConfigReceivers(true, true, true), buildAlertConfigRoute(), buildAlertConfigGlobal(true)),
+					buildAlertConfig(buildAlertConfigReceivers(true, true, true), buildAlertConfigRoute(true), buildAlertConfigGlobal(true)),
```
Comment (joaopalet) [inline stackit/internal/services/argus/argus_acc_test.go:366]: We could add to the comment that we're now also adding child routes

Comment (joaopalet) [reply stackit/internal/services/argus/argus_acc_test.go:366]: `// Update with Alert Configs emailConfigs and global options. Also add child routes` or something

File: stackit/internal/services/argus/argus_acc_test.go [hunk 3]
```diff
@@ -429,7 +464,7 @@ func TestAccResource(t *testing.T) {
 				),
 				// This is needed because the GET request to the alert config endpoint doesn't return some of the global options
 				// and the emailConfigs. Therefore, the state file doesn't contain them and the diff will fail.
-				ExpectNonEmptyPlan: true,
+				// ExpectNonEmptyPlan: true,
```
Comment (joaopalet) [inline stackit/internal/services/argus/argus_acc_test.go:467]: Can be removed, right?

Comment (DiogoFerrao) [reply stackit/internal/services/argus/argus_acc_test.go:467]: Yes, I forgot it there

**Review by joaopalet — APPROVED** [review]

**Pull request merged**